### PR TITLE
Fix TD4 vehicle colour import, move track_design to C++

### PR DESCRIPTION
**Pull request by Gymnasiast**

### CODE REVIEW

File: src/openrct2/ride/TrackDesignRepository.cpp [hunk 1]
```diff
@@ -33,7 +33,7 @@
 
 extern "C"
 {
-    #include "track_design.h"
+    #include "TrackDesign.h"
```
Comment (AaronVanGeffen) [inline src/openrct2/ride/TrackDesignRepository.cpp]: As this now a C++ header file, you should probably remove the `extern C` block.

Comment (Gymnasiast) [reply src/openrct2/ride/TrackDesignRepository.cpp]: It is not _really_ a C++ header file, though. I didn't do anything else than renaming it.

Comment (IntelOrca) [reply src/openrct2/ride/TrackDesignRepository.cpp]: It is still used by other C files so you can't remove it.

File: src/openrct2/ride/TrackDesign.cpp [hunk 2]
```diff
@@ -186,8 +186,10 @@ static rct_track_td6 * track_design_open_from_td4(uint8 * src, size_t srcLength)
         td6->ride_mode = RIDE_MODE_POWERED_LAUNCH;
     }
 
-    // Convert RCT1 vehicle type to RCT2 vehicle type. Intial
-    rct_object_entry vehicleObject = {0x80, {' ', ' ', ' ', ' ', ' ', ' ', ' ', ' '}, 0};
+    // Convert RCT1 vehicle type to RCT2 vehicle type. Intialise with an string consisting of 8 spaces (and no nul terminator).
+    char emptyDatName[9] = "        ";
+    rct_object_entry vehicleObject = {0x80, "", 0};
+    memcpy(vehicleObject.name, emptyDatName, 8);
```
Comment (janisozaur) [inline src/openrct2/ride/TrackDesign.cpp]: This is allocating stack variable and later using this memory to prime the buffer. `memset(vehicleObject.name, ' ', 8);` would suffice.

**Comment by Gymnasiast** [comment]
Anything I still need to do @IntelOrca @janisozaur ?

**Comment by IntelOrca** [comment]
@Gymnasiast lgtm

**Pull request merged**